feat: add bdd test for vault and split the setup from bdd execution

ccojocar opened the pull request:
part of jenkins-x/jx#5190

ccojocar committed:
feat: add bdd test for vault and split the setup from bdd execution

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test all

pmuir commented:
/retest

pmuir commented:
/retest

ccojocar committed:
fix: provide the configuration path to ci script from the pipeline file

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test all

abayer commented on jenkins-x-bdd-vault.yml:
```diff
@@ -0,0 +1,27 @@
+        stages:
+          - name: ci
+            steps:
+              - comamnd: bdd/boot-vault/setup.sh
```
`comamnd` is thoroughly not right. =)

ccojocar committed:
fix: correct typo in the pipeline

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test all

ccojocar committed:
fix: set the proper configuration

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: make the setup script generic and setup the working path for bdd test

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar commented:
/test all

abayer commented on jenkins-x-bdd-vault.yml:
```diff
@@ -21,7 +21,9 @@ pipelineConfig:
               - name: setup
-                command: bdd/boot-vault/setup.sh
+                command: bdd/setup.sh
+                args:['bdd/boot-vault', 'bdd-config']
```
Needs to be a space after the `:`- same in `jenkins-x-bdd-local.yml`.

ccojocar replied:
yap. I think we need to improve the message return when the pipeline runner fails

abayer replied:
Metapipeline helps with that to some extent.

ccojocar committed:
fix: the pipeline files spacing

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: use correct key for working dir

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: use the correct key for working directory

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
chore: add an ls step

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: unify the scripts which set up and run the bdd to avoid changing the working dir between the steps

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

ccojocar committed:
fix: remove the working dir configuration from pipeline files

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

abayer commented:
/lgtm

ccojocar committed:
chore: add abayer into OWNERS file

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

abayer commented:
/lgtm

wbrefvem commented:
/lgtm

ccojocar commented:
/test all

ccojocar committed:
fix: remove the version repo PR option from bdd command arguments

This will make the bdd command to pick the version repository with the commit version from jenkins-x-boot-config repository.

Signed-off-by: Cosmin Cojocar <cosmin.cojocar@gmx.ch>

hferentschik commented:
/lgtm
